internal/node/hybrid/kubelet_cert_validator.go

```diff
@@ -0,0 +1,65 @@
+package hybrid
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/eks-hybrid/internal/kubelet"
+)
+
+// ValidateKubeletCert checks if there is an existing kubelet certificate and validates it against the cluster's CA
+func (hnp *HybridNodeProvider) ValidateKubeletCert() error {
+	hnp.logger.Info("Validating kubelet certificate...")
+
+	certPath := filepath.Join(hnp.installRoot, kubelet.KubeletCurrentCertPath)
+	if _, err := os.Stat(certPath); os.IsNotExist(err) {
+		// No existing cert, validation passes
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("checking kubelet certificate: %v", err)
+	}
+
+	certData, err := os.ReadFile(certPath)
+	if err != nil {
+		return fmt.Errorf("reading kubelet certificate: %v", err)
+	}
+
+	block, _ := pem.Decode(certData)
+	if block == nil {
+		return fmt.Errorf("parsing kubelet certificate")
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("parsing kubelet certificate: %v", err)
+	}
+
+	now := time.Now()
+	if now.Before(cert.NotBefore) {
+		return fmt.Errorf("kubelet server certificate is not yet valid (valid from %s). Verify the system time is correct and remove the kubelet server certificate file %s or use --skip kubelet-cert-validation", cert.NotBefore, certPath)
+	}
+
+	if now.After(cert.NotAfter) {
+		return fmt.Errorf("kubelet server certificate has expired (expired at %s). Please remove the kubelet server certificate file %s", cert.NotAfter, certPath)
+	}
```

Comment:
i'm not sure if these two belong here
if there is a cert but the cert is expired, the kubelet should request a new one. So I don't think init should fail. I'm not sure about a cert in the future tho.